projects/plugins/boost/app/modules/optimizations/image-cdn/class-image-cdn.php

```diff
@@ -18,4 +21,19 @@ public static function get_slug() {
 	public static function is_available() {
 		return true;
 	}
+
+	/**
+	 * Add quality arg to existing photon args.
+	 *
+	 * @param $args array - Existing photon args.
+	 *
+	 * @return mixed
+	 */
+	public function add_quality_args( $args ) {
+		if ( Premium_Features::has_feature( Premium_Features::IMAGE_CDN_QUALITY ) ) {
+			$args['quality'] = jetpack_boost_ds_get( 'image_cdn_quality' );
+		}
```

Comment:
Instead of performing this check on every image, I think this should probably be done at the `setup` level.